File: JWT.Studio/AppScripts/Scripts/Services/EnrollmentService.cs
//opath=F:\projects\JWT.Studio\jwtApp\Scripts\Services\EnrollmentService.js,ab=true
using Jac.Entities.Entities;
using CSharp.Wrapper.Angular;
using CSharp.Wrapper.JS;
using Scripts.Services;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Scripts.Services
{
	[Angular(ModuleName = "app", ActionType = "factory", ActionName = "EnrollmentService", DI = "$http,$q")]
	public class EnrollmentService : BaseService<Enrollment>
	{

		private HttpService http = null;
		private QService qService = null;
		public EnrollmentService(HttpService http, QService qService):base("Enrollment", http, qService)
		{
			this.http=http;
			this.qService=qService;
		}
		public Promise GetCourseList()
		{
			Deferred deffer = this.qService.defer();
			 this.http.get("{0}/GetCourseList".format(this.Root() + this.controllerName))
			.success((res, status) => { deffer.resolve(res); })
			.error((res, status) => { deffer.reject(res); });
			return deffer.promise;
		}
		public Promise GetStudentList()
		{
			Deferred deffer = this.qService.defer();
			 this.http.get("{0}/GetStudentList".format(this.Root() + this.controllerName))
			.success((res, status) => { deffer.resolve(res); })
			.error((res, status) => { deffer.reject(res); });
			return deffer.promise;
		}
	}
}

File: JWT.Studio/jwtApp/Scripts/Services/EnrollmentService.js

(function(window){

var EnrollmentService=Scripts.Services.BaseService.extend({
http:null,
qService:null,
init:function(http,qService)
{
this._super("Enrollment", http, qService);
this.http=http;

this.qService=qService;



},
GetCourseList:function()
{
var deffer=this.qService.defer();
this.http.get("{0}/{1}/GetCourseList".format(this.Root(),this.controllerName)).success(function(res,status){deffer.resolve(res);

}.bind(this)).error(function(res,status){deffer.reject(res);

}.bind(this));


return deffer.promise;


},
GetStudentList:function()
{
var deffer=this.qService.defer();
this.http.get("{0}/{1}/GetStudentList".format(this.Root(),this.controllerName)).success(function(res,status){deffer.resolve(res);

}.bind(this)).error(function(res,status){deffer.reject(res);

}.bind(this));


return deffer.promise;


}});
namespace('Scripts.Services.EnrollmentService',EnrollmentService);

angular.module('app').factory('enrollmentService',['$http','$q',function(http,qService){ return new EnrollmentService(http,qService);}]);
})(window);